src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java

```diff
@@ -13,6 +13,5 @@
 public interface ScoreRepository extends JpaRepository<Score, Score.ScoreId> {
     @Query("SELECT s FROM Score s WHERE s.id = :id")
     Collection<Score> findScoresById(@Param("id") int id);
```

Comment:
1. in method ke kolan estefade nashode. 
2. ham query haye sade ro mishe ba esm method moshakhas kard. spring khodesh mofahme. to intellij `find` ro benevisi baraye esm method khodesh ye list baz mikone komaket mikone
3. method haye sade ro JpaRepository dare to khodesh.mesle finalAll, findById, ...

Reply:
```suggestion
    Collection<Score> findScoresBystudentId(int studentId);
```
intellij nadarm. ba VSCode darm mizanm.
in okye?
badam tu student controller sedash mizanm ba @requestEntity o auth.

Reply:
```suggestion
    List<Score> findScoresBystudentId(int studentId);
```
in okye?
https://stackoverflow.com/a/27627297

Reply:
auth ba UserRole kar mikone. 
baraye bargardondan chandta bayad findAll bezani -> findAllByStudentId
gofftam ke. ti intellij to hmin interface benevis 'find' ye lest barat baz mishe baraye komak. 

method StudentController ham faqat ye RequestEntity niaz dare. sutudentId ro ba auth mishe dar avord

Reply:
intellij nadarm. ba VSCode darm mizanm.